opennms-services/src/main/java/org/opennms/netmgt/poller/QueryManagerDaoImpl.java

```diff
@@ -362,4 +375,36 @@ public void updateLastGoodOrFail(PollableService pollableService, PollStatus sta
         }
     }
 
+    @Override
+    public void persistDeviceConfig(PollableService service, Map<String, Object> attributes, byte[] deviceConfigBytes) {
+
+        // Assume default encoding as ASCII.
+        String encoding = getObjectAsString(attributes.get("encoding"));
+        String  deviceType = getObjectAsString(attributes.get("device-type"));
+        // Retrieve interface
+        final OnmsIpInterface ipInterface = m_ipInterfaceDao.findByNodeIdAndIpAddress(service.getNodeId(), service.getIpAddr());
+        // Fetch last known config for the interface
+        List<DeviceConfig> deviceConfigList = deviceConfigDao.findConfigsForInterfaceSortedByDate(ipInterface);
+        // Update config if it's updated
+        DeviceConfig deviceConfig = deviceConfigList.isEmpty() ? null : deviceConfigList.get(0);
+        if (deviceConfig == null || !Arrays.equals(deviceConfig.getConfig(), deviceConfigBytes)) {
+            deviceConfig = new DeviceConfig();
+            deviceConfig.setConfig(deviceConfigBytes);
+            deviceConfig.setCreatedTime(new Date());
+            deviceConfig.setIpInterface(ipInterface);
+            deviceConfig.setEncoding(encoding);
+            deviceConfig.setDeviceType(deviceType);
+            int version = deviceConfig.getVersion() != null ? deviceConfig.getVersion() + 1 : 1;
+            deviceConfig.setVersion(version);
+            deviceConfigDao.saveOrUpdate(deviceConfig);
+        }
```

Comment:
The loaded instance is no more accessible because `deviceConfig` was reassigned. I propose to use two variables: `loadedDeviceConfig` and `newDeviceConfig`.

```suggestion
        DeviceConfig loadedDeviceConfig = deviceConfigList.isEmpty() ? null : deviceConfigList.get(0);
        if (loadedDeviceConfig == null || !Arrays.equals(loadedDeviceConfig.getConfig(), deviceConfigBytes)) {
            var newDeviceConfig = new DeviceConfig();
            newDeviceConfig.setConfig(deviceConfigBytes);
            newDeviceConfig.setCreatedTime(new Date());
            newDeviceConfig.setIpInterface(ipInterface);
            newDeviceConfig.setEncoding(encoding);
            newDeviceConfig.setDeviceType(deviceType);
            int version = loadedDeviceConfig != null && loadedDeviceConfig.getVersion() != null ? loadedDeviceConfig.getVersion() + 1 : 1;
            newDeviceConfig.setVersion(version);
            deviceConfigDao.saveOrUpdate(newDeviceConfig);
        }
```